src/data_validation.ts

```diff
@@ -1,20 +1,23 @@
 import { MethodRegistrationBase, ArgRequiredOptions } from "./decorators";
-import { DBOSContextImpl } from "./context";
+import { DBOSContextImpl, getCurrentDBOSContext } from "./context";
 import { DBOSDataValidationError } from "./error";
 
 export function validateMethodArgs<Args extends unknown[]>(methReg: MethodRegistrationBase, args: Args)
 {
     let opCtx : DBOSContextImpl | undefined = undefined;
+    if (!methReg.passContext) {
+        opCtx = getCurrentDBOSContext() as DBOSContextImpl;
+    }
 
     const validationError = (msg:string) => {
         const err = new DBOSDataValidationError(msg);
-        opCtx?.span.addEvent("DataValidationError", { message: err.message });
+        opCtx?.span?.addEvent("DataValidationError", { message: err.message });
         return err;
     }
 
     // Input validation
     methReg.args.forEach((argDescriptor, idx) => {
-        if (idx === 0)
+        if (idx === 0 && methReg.passContext)
```

Comment:
Just to understand: what this is doing is gating method arguments validation to operations registered with a context. And this is because we don't have --- and don't see huge benefit in having -- a context free version of `getOrCreateMethodArgsRegistration` ?

Reply:
We're sharing the wrapper.  It needs to know if the context is there or not.  If we did split it we'd be careful not to do too much cut-n-paste?